resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesTaskSetManager.scala

```diff
@@ -52,12 +56,30 @@ private[spark] class KubernetesTaskSetManager(
           if (pendingTasksClusterNodeIP.nonEmpty) {
             logDebug(s"Got preferred task list $pendingTasksClusterNodeIP for executor host " +
               s"$executorIP using cluster node IP $clusterNodeIP")
+            pendingTasksClusterNodeIP
+          } else {
+            val clusterNodeFullName = inetAddressUtil.getFullHostName(clusterNodeIP)
```

Comment:
Nit: I wonder if we can reduce the nesting by returning the pendingTasks ArrayBuffer whenever it's non empty, so we can eliminate all the else blocks in this call.